code/modules/food_and_drinks/drinks/drinks.dm

```diff
@@ -402,11 +402,18 @@
 	isGlass = FALSE
 
 /obj/item/reagent_containers/food/drinks/soda_cans/suicide_act(mob/living/carbon/user)
-	user.visible_message("<span class='suicide'>[user] is trying to eat \the [src]! It looks like [user.p_theyre()] trying to commit suicide!</span>")
-	var/obj/item/trash/can/crushed_can = new /obj/item/trash/can(user.loc)
-	crushed_can.icon_state = icon_state
-	qdel(src)
-	return BRUTELOSS
+	if(!reagents.total_volume)
+		user.visible_message("<span class='warning'>[user] is trying to take a big sip from [src]... The can is empty!</span>")
+		return SHAME
+	if(!is_drainable())
+		open_soda()
+		sleep(10)
+	user.visible_message("<span class='suicide'>[user] takes a big sip from [src]! It looks like [user.p_theyre()] trying to commit suicide!</span>")
+	playsound(user.loc,'sound/items/drink.ogg', 80, 1)
+	reagents.clear_reagents() //a big sip
+	user.say(pick("Now, Outbomb Cuban Pete, THAT was a game.", "They don't make em like Orion Trail anymore.", "You know what they say. Worst day of spess carp fishing is better than the best day at work.", "They don't make em like singularity engines anymore.'"))
+	sleep(20) //dramatic pause
+	return TOXLOSS
```

Comment:
this should check your age, if you're over 40 you survive

Reply:
this is the level of immersion I strive for

Reply:
what would be the best way to handle it? plain `return` just calls the unarmed suicide_act()